### PR TITLE
feat: add capnp

### PR DESCRIPTION
**Pull request by amaanq**
This PR adds a capnp parser+queries

Parser: https://github.com/amaanq/tree-sitter-capnp
Reference: https://capnproto.org/language.html

cc: @marziply since they first brought it up in #2282 

There are some oddities with this language in general that I'd like some advice on as to how to define the queries for. Cap'n Proto fields have a numeric identifier as a type of "version" for that field, e.g. 

```capnp
struct Test {
  voidField    @0  : Void;
  boolField    @1  : Bool;
}
```

So voidField has a version of 0, boolField 1. you can also optionally add a ! to indicate that the number is an inline (immediate) value, as opposed to a reference to a value stored elsewhere in the data structure, for more efficient encoding and decoding of data. 

At first I thought of just using operator, but that wouldn't work and doesn't fit well. I also thought of string.special, but it's not a string at all so that also doesn't fit well. I'm not sure of the use case of `@symbol`, whether if it's just for a special char or if it even applies here. That and the import vs $import are also confusing, I put them both under @include but I wonder if I should just omit $ and instead declare it as `@punctuation.special`.

Thanks!
-Amaan

### CODE REVIEW

File: queries/capnp/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+; Keywords
+
+[
+  "const"
+  "extends"
+  "import"
+  "$import"
+  "namespace"
+  "using"
+] @keyword
```
Comment (theHamsta) [inline queries/capnp/highlights.scm:79]: Isn't `include` already `@include`

**Comment by theHamsta** [comment]
I think the version attributes could be `@attribute`

File: queries/capnp/highlights.scm (hunk 1 above)
Comment (amaanq) [reply queries/capnp/highlights.scm:79]: Not sure what you mean? Do you mean the "import" and "$import" being a duplicate kind of?

Comment (theHamsta) [reply queries/capnp/highlights.scm:79]: I wanted to say that "include" shouldn't be `@keyword` when it's already `@include`

**Comment by amaanq** [comment]
> I think the version attributes could be `@attribute`

Hmm, not the biggest fan of that but that would at least fit without looking awkward, attributes imo feel like an optional note/hint that there's something different about whatever the attribute is covering. The `@number!` fields are always present

File: queries/capnp/highlights.scm (hunk 1 above)
Comment (amaanq) [reply queries/capnp/highlights.scm:77]: fixed

**Review by theHamsta — APPROVED** [review]
This looks great to merge! I'm waiting to see whether someone has a good idea for the version tag/attribute

File: queries/capnp/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+[
+ ; @ ! -
+  "="
+] @operator
+
+; Keywords
+
+[
+  "const"
```
Comment (ObserverOfTime) [inline queries/capnp/highlights.scm:73]: Duplicated in `@type.qualifier`.

**Review by ObserverOfTime — COMMENTED** [review]
> I'm waiting to see whether someone has a good idea for the version tag/attribute

How about `@label` or `@namespace`?

---

The `T` in the following example should be `@type.definition` if possible:

```cpp
using T = Foo.Bar;
```

File: queries/capnp/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  "namespace"
+  "using"
+] @keyword
+
+; Literals
+
+(string_literal) @string
+
+(data_string) @string.special
+(namespace) @string.special
```
Comment (ObserverOfTime) [inline queries/capnp/highlights.scm:84]: Why not `@namespace`?

Comment (amaanq) [reply queries/capnp/highlights.scm:84]: Cap'n Proto namespace declarations are like the following

```capnp
$Cxx.namespace("capnproto_test::capnp::test");

$import "/capnp/c++.capnp".namespace("capnp");
```

I could make the highlight a namespace, but that'd look awkward as it's quoted

File: queries/capnp/highlights.scm (hunk 2 above)
Comment (amaanq) [reply queries/capnp/highlights.scm:73]: > > I'm waiting to see whether someone has a good idea for the version tag/attribute
> 
> How about `@label`?

Yeah, I like label, thanks!

Comment (amaanq) [reply queries/capnp/highlights.scm:73]: fixed const

**Comment by amaanq** [comment]
Could a using directive be `@type.definition`?

i.e.
```capnp
using OuterNestedEnum = TestNestedTypes.NestedEnum;
````

OuterNestedEnum could be `@type.definition`

**Comment by amaanq** [comment]
Need to fix one bug, draft until then

**Comment by amaanq** [comment]
Good now

**Review by ObserverOfTime — APPROVED** [review]

**Comment by amaanq** [comment]
If the namespace query looks good to you @ObserverOfTime, it's ready to merge :)

**Pull request merged**